subsys/bootloader/bl_storage/nrf_nv_counters.c

```diff
@@ -24,22 +24,22 @@ fih_int boot_nv_security_counter_get(uint32_t image_id, fih_int *security_cnt)
 	uint16_t cur_sec_cnt;
 
 	if (security_cnt == NULL) {
-		return FIH_FAILURE;
+		FIH_RET(FIH_FAILURE);
 	}
 
 	/* We support only one counter in MCUBOOT */
 	if (image_id > 0) {
-		return FIH_FAILURE;
+		FIH_RET(FIH_FAILURE);
 	}
 
 	err = get_monotonic_counter(BL_MONOTONIC_COUNTERS_DESC_MCUBOOT_ID0, &cur_sec_cnt);
 	if (err != 0) {
-		return FIH_FAILURE;
+		FIH_RET(FIH_FAILURE);
 	}
 
 	*security_cnt = cur_sec_cnt;
```

Comment:
I think we need to use some fih-macros and functions to encode fih_int here

Reply:
You are right, I will update this